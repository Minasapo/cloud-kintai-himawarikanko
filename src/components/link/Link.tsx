import { createTheme, Link as MuiLink, ThemeProvider } from "@mui/material";

interface LinkProps {
  label?: string;
  href?: string;
  onClick?: () => void;
}

const palette = {
  primary: { main: "#0FA85E", contrastText: "#fff" },
};

const theme = createTheme({
  palette,
  typography: {
    fontFamily: ["Noto Sans JP"].join(","),
  },
  components: {
    MuiCssBaseline: {
      styleOverrides: `
        @font-face {
          font-family: 'Noto Sans JP';
          font-style: bold;
          font-weight: 700;
        }`,
    },
    MuiLink: {
      variants: [
        {
          props: { variant: "button", color: "primary" },
          style: {
            backgroundColor: palette.primary.main,
            color: palette.primary.contrastText,
            textDecoration: "none",
            "&:hover": {
              backgroundColor: palette.primary.contrastText,
              color: palette.primary.main,
              textDecoration: "none",
            },
          },
        },
      ],
    },
  },
});

const Link = ({
  label = "link",
  href = "/",
  onClick = () => {},
  ...props
}: LinkProps) => (
  <ThemeProvider theme={theme}>
    <MuiLink
      href={href}
      variant="button"
      sx={{ p: 1, ...props }}
      onClick={onClick}
    >
      {label}
    </MuiLink>
  </ThemeProvider>
);
export default Link;
