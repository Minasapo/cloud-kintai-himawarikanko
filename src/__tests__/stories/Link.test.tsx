import { render } from "@testing-library/react";

import Link from "../../components/link/Link";

describe("Link", () => {
  test.concurrent("Snapshot", () => {
    const onClickMock = jest.fn();
    const { container } = render(
      <Link label="label" href="/" onClick={onClickMock} />
    );
    expect(container).toMatchSnapshot();
  });
});
