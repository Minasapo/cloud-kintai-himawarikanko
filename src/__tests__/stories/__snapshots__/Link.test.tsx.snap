// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineAlways css-179sywy-MuiTypography-root-MuiLink-root"
    href="/"
  >
    label
  </a>
</div>
`;
